lib/log/log.go

```diff
@@ -0,0 +1,126 @@
+package log
+
+import (
+	"io"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Logger defines a subset of the structured logging interface
+type Logger interface {
```

Comment:
This new interface is supposed to replace the use of `log.FieldLogger` as the latter does not support re-implementations by a custom type. Also, exposes `Writer` API for log streaming.